features/environment.py

```diff
@@ -20,6 +22,30 @@ def read_model(fn, pure):
         model_cache[(fn, pure)] = ifcopenshell.open(fn)
     return model_cache[(fn, pure)]
 
+
+def print_directory_tree(start_path, level=0):
+    """
+    Recursively print the directory tree starting from the given path.
+    """
+    for item in os.listdir(start_path):
+        item_path = os.path.join(start_path, item)
+        print("    " * level + f"|-- {item}")
+        if os.path.isdir(item_path):
+            print_directory_tree(item_path, level + 1)
+
+
+
+def set_logger(context):
+    logger = logging.getLogger("gherkin_rules")
+    if not logger.handlers:
+        base_name = os.path.basename(context.config.userdata.get('input'))
+        file_handler = logging.FileHandler(os.path.join(os.getenv("LOG_FOLDER"), f"gherkin_environment-{os.path.splitext(base_name)[0]}.log"))
```

Comment:
`os.getenv("LOG_FOLDER")` can return None. That should be handled.

Reply:
[Handled](https://github.com/buildingSMART/ifc-gherkin-rules/pull/365/commits/f4946b65a15e9388341a007cba852db2a61d4a96)